### PR TITLE
Added correlation id

### PR DESCRIPTION
**Pull request by vmanilo**
Resolves https://github.com/Twingate/terraform-provider-twingate/issues/338

## Changes
- added header to all requests with correlation id

### CODE REVIEW

File: twingate/provider.go [hunk 1]
```diff
@@ -19,13 +21,19 @@ const (
...
+	EnvNetwork       = "TWINGATE_NETWORK"
+	EnvURL           = "TWINGATE_URL"
+	EnvHTTPTimeout   = "TWINGATE_HTTP_TIMEOUT"
+	EnvHTTPMaxRetry  = "TWINGATE_HTTP_MAX_RETRY"
+	EnvCorrelationID = "TWINGATE_CORRELATION_ID"
 )
 
+func init() { //nolint:gochecknoinits
+	correlationID, _ := uuid.GenerateUUID()
+	_ = os.Setenv(EnvCorrelationID, correlationID)
```
Comment (ekampf) [inline twingate/provider.go:34]: Why do we have different parts of our code communicating a value via OS env var?

This is not a good way and allows interfering with it from the outside

**Comment by vmanilo** [comment]
Initially I tried to init this value directly in http client constructor, but when I run some acc test, I’ve noticed that this id was changing from request to request.I can test some other approach as well to not use env var, but even if someone defines the same env var we will overwrite it.25 мая 2023 г., в 5:40 AM, Eran Kampf ***@***.***> написал(а):﻿
@ekampf commented on this pull request.



In twingate/provider.go:
>  )
 
+func init() { //nolint:gochecknoinits
+	correlationID, _ := uuid.GenerateUUID()
+	_ = os.Setenv(EnvCorrelationID, correlationID)

Why do we have different parts of our code communicating a value via OS env var?
This is not a good way and allows interfering with it from the outside

File: twingate/provider.go (hunk 1 above)
Comment (vmanilo) [reply twingate/provider.go:34]: Initially I tried to init this value directly in http client constructor, but when I run some acc test, I’ve noticed that this id was changing from request to request.I can test some other approach as well to not use env var, but even if someone defines the same env var we will overwrite it.

Comment (vmanilo) [reply twingate/provider.go:34]: I will be back online in a few hours, and can try another approach

Comment (vmanilo) [reply twingate/provider.go]: ok, changed to global var

File: twingate/internal/client/client.go [hunk 2]
```diff
@@ -16,15 +16,19 @@ import (
 	"time"
 
 	"github.com/hashicorp/go-retryablehttp"
+	"github.com/hashicorp/go-uuid"
 	"github.com/hasura/go-graphql-client"
 )
 
+var correlationID, _ = uuid.GenerateUUID() //nolint:gochecknoglobals
```
Comment (ekampf) [inline twingate/internal/client/client.go:23]: Why do we need a global var (and a new lint ignore) when we can just do this in `NewClient` ?
When you create a new client just send it `correlationID: uuid.GenerateUUID()`

Comment (vmanilo) [reply twingate/internal/client/client.go]: fixed

**Review by ekampf — APPROVED** [review]

**Pull request merged**